npm publish changes

NakulManchanda opened the pull request:
Signed-off-by: Nakul Manchanda <nakul.manchanda@ibm.com>

jackjia-ibm commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "@zlux/file-explorer",
-  "version": "0.0.2",
+  "name": "zlux-angular-file-tree",
```
This should be `@zowe/zlux-angular-file-tree` as discussed.

jackjia-ibm commented on tsconfig.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "../../zlux-app-manager/virtual-desktop/plugin-config/tsconfig.base.json",
+  "extends": "./zlux/zlux-app-manager/virtual-desktop/plugin-config/tsconfig.base.json",
```
@NakulManchanda please verify if this will cause any issues on existing build pipeline. Thanks!

NakulManchanda replied:
i think current pipeline will change after this, so should be fine

NakulManchanda commented:
**Summary**:
**Refactoring:**
- New package name:
  **@zowe/zlux-angular-file-tree**
- New Component Selector:
  **zlux-file-tree**
- New Component name
   **ZluxFileTreeComponent**
- New Module Name
  **ZluxFileTreeModule**

**Build Changes:**
**webpack.config.js** 
- changed out folder from `ROOT_DIR/../web` to **`.ROOT_DIR/dist`**
- added sass loader to compile scss files

**tsConfig.json**
- modified extends path to
- "extends": "./zlux/zlux-app-manager/virtual-desktop/plugin-config/tsconfig.base.json",


**Publish changes:**
**package.json** 
- added attribute - **`"main": "dist/main.js"`**
- added dependency @angular/material, 
- added dev dependency  "sass-loader": "~7.1.0",

jackjia-ibm commented:
Hi Nakul, here is summary related to pipeline changes:

- These fields in `package.json` are updated: name, publishConfig, main. Missing packages are added.
- Added Jenkinsfile to include build, dummy test, publish and release stages.
- Lint and audit stages are currently disabled. We should fix this in the future.
- Sonarcloud scan stage is currently disabled temporarily because the branch name published to SonarCloud is not acceptable (when the branch is from a forked repository).
- Added `.npmignore` to skip `zlux` folder to be published. We may need more files be ignored.
